youtube_downloader/downloader.py

```
@@ -258,4 +260,309 @@ def download(self, output_file='video.mp4', itag=None, quality=None):
+                print(f"✓ Skipping {video.get('title', 'video')} (already exists)")
+                if on_video_complete:
+                    on_video_complete(video, output_file)
+                return True
+            
+            # Download the video
+            downloader.download(output_file, quality=quality, itag=itag)
+            
+            if on_video_complete:
+                on_video_complete(video, output_file)
+            
+            return True
```

Ruff (0.14.1)

Consider moving this statement to an `else` block

(TRY300)